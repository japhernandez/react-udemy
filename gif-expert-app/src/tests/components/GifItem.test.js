import React from "react";
import { shallow } from "enzyme";
import GifItem from "../../components/GifItem";

describe("Component <GefItem />", () => {
  const url = "https://gif.com";
  const title = "title gif";
  const wrapper = shallow(<GifItem url={url} title={title} />);
  it("should must show component correctly ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a title within a paragraph", () => {
    const paragraph = wrapper.find("p").text().trim();
    expect(paragraph).toBe(title);
  });

  it("should have a url within a img and alt", () => {
    const image = wrapper.find("img");

    expect(image.prop("src")).toBe(url);
    expect(image.prop("alt")).toBe(title);
  });

  it("should have the class card in div", () => {
    const div = wrapper.find("div");
    const className = div.props().className;
    expect(className.includes("card")).toBe(true);
  });
});
