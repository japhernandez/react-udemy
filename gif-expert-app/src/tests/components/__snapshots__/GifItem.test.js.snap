// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <GefItem /> should must show component correctly  1`] = `
<div
  className="card"
>
  <img
    alt="title gif"
    src="https://gif.com"
  />
  <p>
    title gif
  </p>
</div>
`;
